examples/universal-cnf/vppagent/pkg/ucnf/ucnf_nse.go

```diff
@@ -1,15 +1,94 @@
 package ucnf
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
 	"github.com/danielvladco/k8s-vnet/pkg/nseconfig"
 	"github.com/davecgh/go-spew/spew"
-	"github.com/tiswanso/examples/examples/universal-cnf/vppagent/pkg/config"
+	"github.com/gofrs/uuid"
 	"github.com/networkservicemesh/networkservicemesh/sdk/common"
 	"github.com/sirupsen/logrus"
+	"github.com/tiswanso/examples/api/ipam/ipprovider"
+	"github.com/tiswanso/examples/examples/universal-cnf/vppagent/pkg/config"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
 	"gopkg.in/yaml.v2"
-	"os"
 )
 
+type ipamService struct {
+	IpamAllocator     ipprovider.AllocatorClient
+	registeredSubnets chan *ipprovider.Subnet
+	ctx               *context.Context
+}
+
+func (i *ipamService) AllocateSubnet(ucnfEndpoint *nseconfig.Endpoint) (string, error) {
+	var subnet *ipprovider.Subnet
+	for j := 0; j < 6; j++ {
+		var err error
+		subnet, err = i.IpamAllocator.AllocateSubnet(context.Background(), &ipprovider.SubnetRequest{
+			Identifier: &ipprovider.Identifier{
+				Fqdn:               ucnfEndpoint.CNNS.Address,
+				Name:               ucnfEndpoint.CNNS.Name + uuid.Must(uuid.NewV4()).String(),
+				ConnectivityDomain: ucnfEndpoint.CNNS.ConnectivityDomain,
+			},
+			AddrFamily: &ipprovider.IpFamily{Family: ipprovider.IpFamily_IPV4},
+			PrefixLen:  uint32(ucnfEndpoint.VL3.IPAM.PrefixLength),
+		})
+		if err != nil {
+			if j == 5 {
+
+			}
+			logrus.Errorf("ipam allocation not successful: %v \n waiting 60 seconds before retrying \n", err)
```

Comment:
This statement will be shown even when the final loop executes.  Maybe try to use the above if statement above to do something about this.

Reply:
done